docs/pages/architecture/tls-routing.mdx

```diff
@@ -157,6 +148,34 @@ Teleport proxy's web listener and don't require a local proxy or any special
 ALPN/SNI negotiation. These web connections use standard `http1.1` and `h2`
 protocols for ALPN.
 
+Application access for TCP applications connects through a local proxy by `tsh
+proxy app` where the local proxy forwards the traffic to Teleport Proxy service
+using the `teleport-tcp` ALPN protocol.
+
+## Working with layer 7 load balancers or reverse proxies (Preview)
+
+Starting from version `13.0`, TLS routing can now be enabled allowing the
+Teleport proxy to serve a single port behind a layer 7 load balancer or reverse
+proxy.
+
+![Layer 7 load balancer setup](../../img/architecture/tls-routing-alb.svg)
+
+It is expected that the layer 7 load balancer or reverse proxy will terminate
+TLS with a public certificate, such as using ACM for AWS ALB. This means that
+the Proxy service does not require a Web TLS certificate using `http_keypair`
+or `acme`.
+
+Teleport clients automatically detect whether Teleport proxy behind a layer 7
+load balancer or a reverse proxy. In such cases, the client initiates a
+connection upgrade similar to "WebSocket" to bypass the load balancer and then
```

Comment:
```suggestion
load balancer or a reverse proxy. In such cases, the client initiates a
HTTP Connection Upgrade to bypass the load balancer and then
```
Also  "bypass"  in this context might be misunderstand we don't bypass the load balance per-se. 
Maybe "to forward TLS handshake to the Teleport Proxy" ?